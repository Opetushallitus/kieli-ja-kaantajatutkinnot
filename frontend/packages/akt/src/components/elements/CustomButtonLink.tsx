import { Button, ButtonProps } from '@mui/material';
import { FC } from 'react';
import { Link, LinkProps } from 'react-router-dom';

type CustomButtonLinkProps = ButtonProps &
  Pick<LinkProps, 'target' | 'rel' | 'to'>;

export const CustomButtonLink: FC<CustomButtonLinkProps> = ({
  to,
  target,
  rel,
  ...btnProps
}) => {
  return (
    <Link className="custom-button-link" to={to} target={target} rel={rel}>
      <Button {...btnProps}></Button>
    </Link>
  );
};
