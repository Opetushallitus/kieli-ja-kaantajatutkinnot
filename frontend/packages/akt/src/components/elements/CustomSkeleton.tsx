import { Skeleton, SkeletonProps } from '@mui/material';
import { FC } from 'react';

import { useCommonTranslation } from 'configs/i18n';

export const CustomSkeleton: FC<SkeletonProps> = ({ ...props }) => {
  const translateCommon = useCommonTranslation();

  return <Skeleton {...props} aria-label={translateCommon('loadingContent')} />;
};
