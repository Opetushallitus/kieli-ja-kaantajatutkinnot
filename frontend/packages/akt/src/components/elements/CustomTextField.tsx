import { TextField } from '@mui/material';

import { CustomTextFieldProps } from 'interfaces/components/customTextField';

export const CustomTextField = ({
  error,
  helperText,
  showHelperText,
  multiline,
  ...rest
}: CustomTextFieldProps) => {
  const minRows = multiline ? 5 : undefined;
  const maxRows = multiline ? 15 : undefined;

  return (
    <TextField
      minRows={minRows}
      maxRows={maxRows}
      error={error}
      multiline={multiline}
      helperText={(error || showHelperText) && helperText}
      {...rest}
    />
  );
};
