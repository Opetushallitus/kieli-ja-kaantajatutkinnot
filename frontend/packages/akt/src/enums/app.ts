export enum AppConstants {
  CallerID = '1.2.246.562.10.00000000001.akt',
}

export enum AppRoutes {
  PublicHomePage = '/akt/etusivu',
  ClerkHomePage = '/akt/virkailija',
  MeetingDatesPage = '/akt/virkailija/kokouspaivat',
  ClerkSendEmailPage = '/akt/virkailija/laheta-sahkoposti',
  ClerkLocalLogoutPage = '/akt/cas/localLogout',
  ClerkTranslatorOverviewPage = '/akt/virkailija/kaantaja/:translatorId',
  ClerkNewTranslatorPage = '/akt/virkailija/lisaa-kaantaja',
  AccessibilityStatementPage = '/akt/saavutettavuusseloste',
  NotFoundPage = '*',
}

export enum Duration {
  Short = 3000,
  Medium = 6000,
  MediumExtra = 9000,
  Long = 12000,
}

export enum SearchFilter {
  FromLang = 'fromLang',
  ToLang = 'toLang',
  Name = 'name',
  Town = 'town',
}

export enum KeyboardKey {
  Enter = 'Enter',
}

export enum PublicUIViews {
  PublicTranslatorListing = 'PublicTranslatorListing',
  ContactRequest = 'ContactRequest',
}

export enum TextFieldTypes {
  Text = 'text',
  Email = 'email',
  PhoneNumber = 'tel',
  Textarea = 'textarea',
  PersonalIdentityCode = 'personalIdentityCode',
}

export enum CustomTextFieldErrors {
  Required = 'errors.customTextField.required',
  MaxLength = 'errors.customTextField.maxLength',
  EmailFormat = 'errors.customTextField.emailFormat',
  TelFormat = 'errors.customTextField.telFormat',
  PersonalIdentityCodeFormat = 'errors.customTextField.personalIdentityCodeFormat',
}

export enum NotifierTypes {
  Dialog = 'dialog',
  Toast = 'toast',
}

export enum Severity {
  Info = 'info',
  Success = 'success',
  Error = 'error',
  Warning = 'warning',
}

export enum Variant {
  Text = 'text',
  Outlined = 'outlined',
  Contained = 'contained',
}

export enum TextFieldVariant {
  Filled = 'filled',
  Outlined = 'outlined',
  Standard = 'standard',
}

export enum SkeletonVariant {
  Text = 'text',
  Rectangular = 'rectangular',
  Circular = 'circular',
}

export enum I18nNamespace {
  Common = 'common',
  Translation = 'translation',
  KoodistoLanguages = 'koodistoLanguages',
  Accessibility = 'accessibility',
}

export enum Color {
  Primary = 'primary',
  Secondary = 'secondary',
  Inherit = 'inherit',
  Error = 'error',
  Warning = 'warning',
}

export enum HTTPStatusCode {
  Ok = 200,
  Created = 201,
  Accepted = 202,
  NoContent = 204,
  MovedPermanently = 301,
  Found = 302,
  NotModified = 304,
  TemporaryRedirect = 307,
  PermanentRedirect = 308,
  BadRequest = 400,
  Unauthorized = 401,
  Forbidden = 403,
  NotFound = 404,
  MethodNotAllowed = 405,
  InternalServerError = 500,
  NotImplemented = 501,
  BadGateway = 502,
  ServiceUnavailable = 503,
  GatewayTimeout = 504,
}

export enum Screenwidth {
  Phone = 480,
  Tablet = 800,
  Desktop = 1024,
}

export enum PermissionToPublish {
  Yes = 'Kyll√§',
  No = 'Ei',
}

export enum AppLanguage {
  Finnish = 'fi-FI',
  Swedish = 'sv-SE',
  English = 'en-GB',
}

export enum Direction {
  Horizontal = 'horizontal',
  Vertical = 'vertical',
}

export enum UIMode {
  View = 'view',
  EditTranslatorDetails = 'editTranslatorDetails',
  EditAuthorisationDetails = 'editAuthorisationDetails',
}

export enum HeaderTabNav {
  Register = 'register',
  MeetingDates = 'meetingDates',
}
