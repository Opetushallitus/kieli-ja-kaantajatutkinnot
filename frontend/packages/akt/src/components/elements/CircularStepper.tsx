import { CircularProgress, CircularProgressProps } from '@mui/material';

import { H3 } from 'components/elements/Text';
import { useCommonTranslation } from 'configs/i18n';
import { Color } from 'enums/app';

interface CircularStepperProps extends CircularProgressProps {
  value: number;
  phaseText: string;
  phaseDescription: string;
}

export const CircularStepper = ({
  phaseText,
  phaseDescription,
  ...rest
}: CircularStepperProps) => {
  const translateCommon = useCommonTranslation();
  const ariaLabel = `${translateCommon(
    'phase'
  )} ${phaseText}: ${phaseDescription}`;

  return (
    <div className="circular-stepper">
      <CircularProgress
        aria-label={ariaLabel}
        className="circular-stepper__progress"
        color={Color.Secondary}
        variant="determinate"
        {...rest}
      />
      <div className="circular-stepper__heading">
        <H3>{phaseText}</H3>
      </div>
    </div>
  );
};
