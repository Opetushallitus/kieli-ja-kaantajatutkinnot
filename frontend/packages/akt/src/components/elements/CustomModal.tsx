import Modal, { ModalProps } from '@mui/material/Modal';
import { FC } from 'react';

import { H2 } from 'components/elements/Text';

type CustomModalProps = ModalProps & {
  ariaLabelledBy?: string;
  screenReaderTitle?: string;
  modalTitle?: string;
  onCloseModal: () => void;
};

export const CustomModal: FC<CustomModalProps> = ({
  ariaLabelledBy,
  open,
  children,
  modalTitle,
  onCloseModal,
}) => {
  const handleOnClose = (event: React.SyntheticEvent, reason: string) => {
    if (event && reason === 'backdropClick') {
      return;
    }
    onCloseModal();
  };

  return (
    <Modal open={open} onClose={handleOnClose} aria-labelledby={ariaLabelledBy}>
      <div className="custom-modal">
        <div className="rows gapped">
          <H2 id={ariaLabelledBy}>{modalTitle}</H2>
          {children}
        </div>
      </div>
    </Modal>
  );
};
