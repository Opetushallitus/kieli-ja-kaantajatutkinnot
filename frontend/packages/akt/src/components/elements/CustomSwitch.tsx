import {
  FormControlLabel,
  FormGroup,
  Switch,
  SwitchProps,
} from '@mui/material';

import { Caption, Text } from 'components/elements/Text';
import { Color } from 'enums/app';

interface CustomSwitchProps extends SwitchProps {
  dataTestId?: string;
  leftLabel: string;
  rightLabel: string;
  errorLabel?: string;
  value?: boolean;
}

export const CustomSwitch = ({
  dataTestId,
  leftLabel,
  rightLabel,
  errorLabel,
  value,
  disabled,
  onChange,
}: CustomSwitchProps) => {
  const leftLabelClassName = disabled
    ? 'color-disabled margin-right-xs'
    : 'margin-right-xs';

  return (
    <FormGroup className="rows">
      <div className="columns">
        <Text className={leftLabelClassName}>{leftLabel}</Text>
        <FormControlLabel
          disabled={disabled}
          control={
            <Switch
              data-testid={dataTestId}
              checked={value}
              color={Color.Secondary}
              onChange={onChange}
            />
          }
          label={rightLabel}
        />
      </div>
      {errorLabel && (
        <Caption data-testid={`${dataTestId}__error-label`} color={Color.Error}>
          {errorLabel}
        </Caption>
      )}
    </FormGroup>
  );
};
