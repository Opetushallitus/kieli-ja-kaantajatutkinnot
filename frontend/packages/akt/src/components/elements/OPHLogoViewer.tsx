import { FC } from 'react';

import { Svg } from 'components/elements/Svg';
import { getCurrentLang, useCommonTranslation } from 'configs/i18n';
import { AppLanguage, Direction } from 'enums/app';
import OPHLogoENHorizontal from 'public/assets/svg/oph_logo_horiz_en.svg';
import OPHLogoHorizontal from 'public/assets/svg/oph_logo_horiz_fi_sv.svg';
import OPHLogoENVertical from 'public/assets/svg/oph_logo_vert_en.svg';
import OPHLogoVertical from 'public/assets/svg/oph_logo_vert_fi_sv.svg';

interface OPHLogoViewerProps {
  direction: Direction;
  className?: string;
}

export const OPHLogoViewer: FC<OPHLogoViewerProps> = ({
  className,
  direction,
}) => {
  const currentLang = getCurrentLang();
  const translateCommon = useCommonTranslation();

  const getLogo = () => {
    const isEnglish = currentLang === AppLanguage.English;

    if (isEnglish && direction === Direction.Horizontal) {
      return OPHLogoENHorizontal;
    } else if (isEnglish && direction === Direction.Vertical) {
      return OPHLogoENVertical;
    } else if (!isEnglish && direction === Direction.Horizontal) {
      return OPHLogoHorizontal;
    } else if (!isEnglish && direction === Direction.Vertical) {
      return OPHLogoVertical;
    }
  };

  return (
    <Svg
      className={className}
      src={getLogo()}
      alt={translateCommon('ophLogo')}
    />
  );
};
