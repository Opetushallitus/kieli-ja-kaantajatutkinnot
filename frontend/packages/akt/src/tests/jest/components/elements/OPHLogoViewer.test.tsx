import renderer from 'react-test-renderer';

import { OPHLogoViewer } from 'components/elements/OPHLogoViewer';
import { Direction } from 'enums/app';

describe('OPHLogoViewer', () => {
  it('should render OPHLogoViewer correctly', () => {
    const tree = renderer
      .create(<OPHLogoViewer direction={Direction.Horizontal} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
