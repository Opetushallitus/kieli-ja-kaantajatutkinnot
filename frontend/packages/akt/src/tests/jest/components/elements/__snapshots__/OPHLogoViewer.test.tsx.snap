// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OPHLogoViewer should render OPHLogoViewer correctly 1`] = `
<img
  alt="ophLogo"
  src="test"
/>
`;
